apps/sms/test/marionette/drafts_test.js

```diff
@@ -145,51 +186,240 @@ marionette('Messages Drafts', function() {
+      var newMessageView, originalDraftConversation;
 
-      createAndSaveDraft(draft);
+      setup(function() {
+        newMessageView = inboxView.createNewMessage();
+        originalDraftConversation = createAndSaveDraft(newMessageView, draft);
 
-      assertDraft(draft);
+        assertConversation(originalDraftConversation, {
+          isDraft: true,
+          lastMessageType: 'mms',
+          title: draft.recipients[0],
+          bodyText: ''
+        });
 
-      // Relaunch application and verify draft is persisted.
-      messagesApp.close();
-      messagesApp.launch();
-      storage.setMessagesStorage(messagesStorage);
+        inboxView.goToConversation(originalDraftConversation.id);
+        assertDraft(newMessageView, draft);
+      });
+
+      test('Draft is correctly saved', function() {
+        // Relaunch application and verify draft is persisted.
+        relaunchApp();
+
+        assertConversation(
+          inboxView.conversations[0], originalDraftConversation
+        );
+
+        inboxView.goToConversation(originalDraftConversation.id);
+        assertDraft(newMessageView, draft);
+      });
+
+      test('Draft is correctly modified', function() {
+        var newDraft = {
+          text: 'content_changed',
+          recipients: ['+234']
+        };
+
+        newMessageView.clearRecipients();
+        newMessageView.clearMessage();
+        newMessageView.hideSubject();
+
+        var updatedDraftConversation = createAndSaveDraft(
+          newMessageView, newDraft
+        );
+
+        assertConversation(updatedDraftConversation, {
+          id: originalDraftConversation.id,
+          isDraft: true,
+          lastMessageType: 'sms',
+          title: newDraft.recipients[0],
+          bodyText: newDraft.text
+        });
+
+        inboxView.goToConversation(updatedDraftConversation.id);
+        assertDraft(newMessageView, newDraft);
+
+        // Relaunch application and verify draft is persisted.
+        relaunchApp();
+
+        assertConversation(
+          inboxView.conversations[0], updatedDraftConversation
+        );
+
+        inboxView.goToConversation(updatedDraftConversation.id);
+        assertDraft(newMessageView, newDraft);
+      });
+
+      test('Draft is correctly discarded', function() {
+        var originalConversation = discardDraft(newMessageView, draft);
 
-      assertDraft(draft);
+        assertConversation(originalConversation, {
+          id: conversation.id,
+          title: conversation.participants[0],
+          bodyText: conversation.body,
+          lastMessageType: 'sms',
+          isDraft: false,
+          hasDraft: false
+        });
+
+        // Relaunch application and verify draft is persistently discarded.
+        relaunchApp();
+
+        var conversations = inboxView.conversations;
+        assert.equal(conversations.length, 1);
+        assertConversation(conversations[0], originalConversation);
+      });
```

Comment:
wondering if you could merge these 3 tests in 1, so that we avoid restarting the full B2G Desktop and save time ?